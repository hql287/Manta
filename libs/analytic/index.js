const {
  getQueue,
  setQueue,
  cleanQueue,
  processQueue,
  getCurrentSessionHits,
  setCurrentSessionHits,
} = require('./q');
const { getNow, getDefaultParams, dispatch } = require('./helpers');

class Analytics {
  constructor(trackingID, clientID, options = {}) {
    if (!clientID) throw new Error('clientID is required');
    if (!trackingID) throw new Error('trackingID is required');
    this.initialParams = Object.assign({}, options, getDefaultParams(), {
      tid: trackingID,
      cid: clientID,
    });
    this.currentSessionHits = [];
    this.init();
  }

  async init() {
    const unsentSessions = getQueue();
    const currentSessionHits = getCurrentSessionHits();
    const q = [...unsentSessions, currentSessionHits];
    if (q.length === 0) return;
    const q1 = await processQueue(q);
    const q2 = cleanQueue(q1);
    setQueue(q2);
    setCurrentSessionHits([]);
  }

  track(hitType, hitParams) {
    const payload = this._getPayload(hitType, hitParams, getNow());
    dispatch(payload)
      .then(res => {
        console.log(res);
      })
      .catch(err => {
        this.currentSessionHits.push(payload);
        setCurrentSessionHits(this.currentSessionHits);
      });
  }

  _getPayload(hitType, hitParams = {}, time) {
    return Object.assign(
      {},
      this.initialParams, // Initial params
      hitParams, // Additional hit params
      {
        t: hitType, // Hit type
        dp: hitParams.cd, // Screen name
        __timeStamp: time, // Use to calculate queueTime later
      }
    );
  }
}

module.exports = Analytics;
