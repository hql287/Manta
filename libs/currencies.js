import currencies from './currencies.json';
import cryptocurrencies from 'cryptocurrencies';

const Crypto = () => {
    let cryptocurrencie = {};

    const keys = Object.keys(cryptocurrencies)

    for (let index = 0; index < keys.length; index++) {
        
        cryptocurrencie[keys[index]] = {
            "symbol": keys[index],
            "name":  cryptocurrencies[keys[index]],
            "symbol_native": keys[index],
            "decimal_digits": 2,
            "rounding": 0,
            "code": keys[index],
            "name_plural": ""
        }
    }

    return cryptocurrencie;
}

const All = () => (Object.assign({}, currencies, Crypto()))

const List = () => ({ currencies, cryptocurrencies: Crypto()});

export {
    All,
    List
}