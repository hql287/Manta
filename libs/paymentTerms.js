export const paymentTerms = ['net7', 'net10', 'net30', 'net60', 'net90'];
