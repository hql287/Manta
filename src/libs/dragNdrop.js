// Prevent window to open dropped file
document.addEventListener('dragover', event => event.preventDefault());
document.addEventListener('drop', event => event.preventDefault());

// Linux and Windows draggable window broken, so monkey patch
if (process.platform == 'linux' || process.platform == 'win32') {
  const elements = document.querySelectorAll('body,#root,.appWrapper');
  elements.forEach(element => {
    element.classList.add('non-draggable');
  });
}
