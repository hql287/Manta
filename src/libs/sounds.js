// 3rd Party Libs
const appConfig = require('electron').remote.require('electron-settings');

// Sound Themes
const cs_sounds = require('../static/sounds/cs/index.js');
const default_sounds = require('../static/sounds/default/index.js');

let cache;
function preload() {
  cache = {};
  setSounds();
  for (const name in sounds) {
    if (!cache[name]) {
      const sound = sounds[name];
      const audio = (cache[name] = new window.Audio());
      audio.volume = sound.volume;
      audio.src = sound.url;
    }
  }
}

let sounds;
function setSounds() {
  const soundTheme = appConfig.get('general.sound');
  switch (soundTheme) {
    case 'cs': {
      sounds = cs_sounds;
      break;
    }
    default: {
      sounds = default_sounds;
      break;
    }
  }
}

function play(name) {
  const appMute = appConfig.get('general.muted');
  if (!appMute) {
    let audio = cache[name];
    if (!audio) {
      const sound = sounds[name];
      if (!sound) {
        throw new Error('Invalid sound name');
      }
      audio = cache[name] = new window.Audio();
      audio.volume = sound.volume;
      audio.src = sound.url;
    }
    audio.currentTime = 0;
    audio.play();
  }
}

module.exports = {
  preload,
  play,
};
