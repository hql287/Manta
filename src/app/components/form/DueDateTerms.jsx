// Libraries
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Styles
import _withFadeInAnimation from '../shared/hoc/_withFadeInAnimation';
import styled from 'styled-components';
const Container = styled.div`
  display: flex;
  margin-bottom: 20px;
`;

// Payment Terms
import { paymentTerms } from '../../../libs/paymentTerms';

// Component
export class DueDateTerms extends Component {
  constructor(props) {
    super(props);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  componentWillMount() {
    if (this.props.paymentTerm === null) {
      this.updateSelectedTerm(paymentTerms[0]);
    }
  }

  handleInputChange(event) {
    this.updateSelectedTerm(event.target.value);
  }

  updateSelectedTerm(term) {
    const { updatePaymentTerm } = this.props;
    updatePaymentTerm(term);
  }

  render() {
    const { t, paymentTerm } = this.props;
    const termOptions = paymentTerms.map(term => (
      <option key={term} value={term}>
        { t(`form:fields:dueDate:paymentTerms:${term}:label`) }
        {' - '}
        { t(`form:fields:dueDate:paymentTerms:${term}:description`) }
      </option>
    ));
    return (
      <Container>
        <select
          onChange={this.handleInputChange}
          value={paymentTerm === null ? paymentTerms[0] : paymentTerm}
        >
          { termOptions }
        </select>
      </Container>
    );
  }
}

DueDateTerms.propTypes = {
  paymentTerm: PropTypes.string,
  t: PropTypes.func.isRequired,
  updatePaymentTerm: PropTypes.func.isRequired,
};

// Export
export default _withFadeInAnimation(DueDateTerms);

