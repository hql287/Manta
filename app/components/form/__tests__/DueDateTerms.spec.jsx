// Libs
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

// Component
import { DueDateTerms } from '../DueDateTerms.jsx';
import { paymentTerms } from '../../../../libs/paymentTerms';

// Mocks
const t = jest.fn();
const updatePaymentTerm = jest.fn();
const paymentTerm = 'net10';

describe('Renders correctly to the DOM', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <DueDateTerms t={t} paymentTerm={paymentTerm} updatePaymentTerm={updatePaymentTerm} />
    );
  });

  // TODO
  it('will not rerender if threre is no change in Props', () => {});

  it('receives correct props', () => {
    expect(wrapper.prop('paymentTerm')).toEqual(paymentTerm);
    expect(wrapper.prop('paymentTerm')).not.toEqual({});
    expect(wrapper.prop('updatePaymentTerm')).toEqual(updatePaymentTerm);
  });

  it('renders necessary element', () => {
    expect(wrapper.find('select')).toHaveLength(1);
    expect(wrapper.find('select')).not.toHaveLength(2);
    expect(wrapper.find('option')).toHaveLength(paymentTerms.length);
  });

  it('handle input change correctly', () => {
    // Setup
    const spy = jest.spyOn(DueDateTerms.prototype, 'handleInputChange');
    const wrap = mount(
      <DueDateTerms t={t} paymentTerm={paymentTerm} updatePaymentTerm={updatePaymentTerm} />
    );
    // Execute
    const select = wrap.find('select');
    select.simulate('change', { target: { value: 'net90' } });
    // Assertion
    expect(spy).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <DueDateTerms t={t} paymentTerm={paymentTerm} updatePaymentTerm={updatePaymentTerm} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
