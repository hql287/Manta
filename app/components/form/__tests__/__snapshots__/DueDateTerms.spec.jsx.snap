// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly to the DOM matches snapshot 1`] = `
<div
  className="DueDateTerms__Container-grvKkj cHwqdm"
>
  <select
    onChange={[Function]}
    value="net10"
  >
    <option
      value="net7"
    >
       - 
    </option>
    <option
      value="net10"
    >
       - 
    </option>
    <option
      value="net30"
    >
       - 
    </option>
    <option
      value="net60"
    >
       - 
    </option>
    <option
      value="net90"
    >
       - 
    </option>
  </select>
</div>
`;
